doc/case_studies/exploring-weather.rst

```diff
@@ -226,33 +226,33 @@ of the selection (for more information on selections, see
 .. altair-plot::
 
     brush = alt.selection_interval()
-
-    points = alt.Chart().mark_point().encode(
-        alt.X('temp_max:Q').title('Maximum Daily Temperature (C)'),
-        alt.Y('temp_range:Q').title('Daily Temperature Range (C)'),
-        color=alt.condition(brush, 'weather:N', alt.value('lightgray'), scale=scale),
-        size=alt.Size('precipitation:Q').scale(range=[1, 200])
-    ).transform_calculate(
-        "temp_range", "datum.temp_max - datum.temp_min"
-    ).properties(
-        width=600,
-        height=400
-    ).add_params(
-        brush
+    color = (
+        alt.when(brush)
+        .then(alt.Color("weather:N").scale(scale))
+        .otherwise(alt.value("lightgray"))
     )
 
-    bars = alt.Chart().mark_bar().encode(
-        x='count()',
-        y='weather:N',
-        color=alt.Color('weather:N').scale(scale),
-    ).transform_calculate(
-        "temp_range", "datum.temp_max - datum.temp_min"
-    ).transform_filter(
-        brush
-    ).properties(
-        width=600
+    points = (
+        alt.Chart()
+        .mark_point()
+        .encode(
+            alt.X("temp_max:Q").title("Maximum Daily Temperature (C)"),
+            alt.Y("temp_range:Q").title("Daily Temperature Range (C)"),
+            color=color,
+            size=alt.Size("precipitation:Q").scale(range=[1, 200]),
+        )
+        .transform_calculate("temp_range", "datum.temp_max - datum.temp_min")
+        .properties(width=600, height=400)
+        .add_params(brush)
+    )
+    bars = (
+        alt.Chart()
+        .mark_bar()
+        .encode(x="count()", y="weather:N", color=alt.Color("weather:N").scale(scale))
+        .transform_calculate("temp_range", "datum.temp_max - datum.temp_min")
+        .transform_filter(brush)
+        .properties(width=600)
```

Comment:
Happy to discuss in another issue, I shared a similar opinion to that of @binste in https://github.com/vega/altair/pull/3485#issuecomment-2240025423

Specific to this PR - I won't oppose any formatting changes you make, but I would prefer to leave that until review.
I'm also pasting in from `ruff`, so these style choices are inherited from https://github.com/vega/altair/blob/main/pyproject.toml

Reply:
Sounds good, I opened a discussion in https://github.com/vega/altair/discussions/3570

Reply:
@joelostblom I'm "resolving" this since I don't want it to be the focus for review

Please know that I'm not dismissing formatting changes.

As I said in https://github.com/vega/altair/pull/3544#discussion_r1745587004, feel free to make those directly! 👍